sklearn/metrics/tests/test_classification.py

```diff
@@ -2530,6 +2530,15 @@ def test_log_loss():
     loss = log_loss(y_true, y_score2, labels=[1, 2, 3])
     assert_almost_equal(loss, 1.0630345, decimal=6)
 
+    # ensure np.float32 and np.float16 inputs give correct output
+    for dtype in [np.float16, np.float32]:
+        y_true = [1, 2, 2]
+        y_score2 = np.array(
+            [[0.2, 0.7, 0.3], [0.6, 0.5, 0.3], [0.3, 0.9, 0.1]], dtype=dtype
+        )
+        loss = log_loss(y_true, y_score2, labels=[1, 2, 3])
+        assert_almost_equal(loss, 1.0630345, decimal=3)
```

Comment:
Several things here:

- We try to find an example that is not really specific. Here, relying on so many decimal is problematic.
- If we want to check for floating numbers, we should use `assert_allclose` instead of `almost_equal`.

In the original issue, it was shown that the result would be `np.nan` if I am not wrong. We could retake the same arrays and make sure that the output values is finite.

This would have failed in `main` but would pass the test with the current fix.